main/accounts/index.ts

```diff
@@ -523,16 +524,27 @@ export class Accounts extends EventEmitter {
     const request = currentAccount.requests[handlerId] && currentAccount.requests[handlerId].type === 'transaction'
     if (!request) return cb(new Error(`Could not locate request ${handlerId}`))
 
+    const lastSignerType = getSignerType(currentAccount.lastSignerType)
     const signer = currentAccount.getSigner()
-    if (!signer) return cb(new Error('No signer'))
-
-    if (signer.status === 'locked') {
-      const crumb = {
-        view: 'expandedSigner', 
-        data: { signer: signer.id }
+    const isHardware = isHardwareSigner(lastSignerType)
+    
+    // handle locked and other states requiring user action
+    if (isHardware || signer) {
```

Comment:
@goosewobbler this should be `isHardware && !signer` right?

Reply:
And then we also need a case for: `signer && signer.status !== 'ok'`

Reply:
No because it covers hot signers too, I integrated the bit from below for the hot signers